kombu/transport/redis.py

```diff
@@ -1288,6 +1288,9 @@ def __init__(self, *args, **kwargs):
         # All channels share the same poller.
         self.cycle = MultiChannelPoller()
 
+        # tracks event loop (on_tick) entries by connection
+        self.on_poll_start_by_connection = {}
```

Comment:
are we going to undo https://github.com/celery/kombu/pull/1476 by this changes or adjusting that? can you add integration tests as well?

Reply:
> are we going to undo https://github.com/celery/kombu/pull/1476 by this changes or adjusting that?

This is an adjustment since it will still remove functions from `on_tick` after disconnection occurs to prevent memory leaks, but now we'll be making sure it removes the correct function. I can do some of the same tests I did in https://github.com/celery/kombu/pull/1476 to make sure the leak didn't come back.

> can you add integration tests as well?

I'll work on adding that integration test.

Reply:
OK looking forward to that